sympy/functions/elementary/complexes.py

```diff
@@ -320,6 +320,8 @@ class sign(Function):
     def doit(self, **hints):
         if self.args[0].is_zero is False:
             return self.args[0] / Abs(self.args[0])
+        if self.args[0].is_zero is True:
```

Comment:
Would it be better to just call `eval`? I can imagine many other cases not really working for `Sign(..., evaluate=False).doit()`.  `oo` will give `NaN` instead of 1 etc. So maybe try to evaluate it first (in case it use `evaluate=False` and if that results the same answer, do lines 321-322?

Reply:
Or maybe just first call `doit` for the super class. It seems like `sign` is one of the very few standard functions that has its own `doit`. Most others are `Integral` etc, where a significant amount of work is required, not just evaluating.

Reply:
If the super `doit` method calls `eval` then presumably this `doit` method can just be deleted.

Reply:
This `doit` does the rewrite which the normal `eval` doesn't do as far as I understand. I may be wrong though.

Reply:
I think if `eval` handles the case then a `Function` subclass should not define `doit`:
```python
In [2]: sign.eval(S.Zero)
Out[2]: 0

In [3]: class A(Function):
   ...:     @classmethod
   ...:     def eval(cls, arg):
   ...:         return pi
   ...: 

In [4]: A(1)
Out[4]: π

In [5]: A(1, evaluate=False)
Out[5]: A(1)

In [6]: A(1, evaluate=False).doit()
Out[6]: π
```